src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -520,13 +521,11 @@ public CodeFixProvider GetSuppressionFixer(string language, IEnumerable<string>
 
                     // Have to see if this fix is still applicable.  Jump to the foreground thread
                     // to make that check.
-                    var applicable = await Task.Factory.StartNew(() =>
-                        {
-                            this.AssertIsForeground();
-                            return fix.Action.IsApplicable(document.Project.Solution.Workspace);
-                        },
-                        cancellationToken, TaskCreationOptions.None, ForegroundTaskScheduler).ConfigureAwait(false);
-                    this.AssertIsBackground();
+                    await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
❕ This needs to yield if called from the main thread in order to preserve the original behavior.